exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/SetOptionHandler.java

```diff
@@ -113,6 +113,12 @@ public PhysicalPlan getPlan(SqlNode sqlNode) throws ValidationException, Foreman
       }
     }
 
+    // Skip writing profiles for "ALTER SESSION SET" queries
+    if (optionScope == OptionScope.SESSION && options.getBoolean(ExecConstants.SKIP_ALTER_SESSION_QUERY_PROFILE)) {
+      logger.debug("Will not write profile for ALTER SESSION SET ... ");
+      context.skipWritingProfile(true);
```

Comment:
this logic can be moved to line 73. No need to again check for `optionScope` here.